lib/Scheduler/aco.cpp

```diff
@@ -202,13 +242,13 @@ std::unique_ptr<InstSchedule> ACOScheduler::FindOneSchedule() {
   std::unique_ptr<InstSchedule> schedule =
       llvm::make_unique<InstSchedule>(machMdl_, dataDepGraph_, true);
   InstCount maxPriority = rdyLst_->MaxPriority();
-  if (maxPriority == 0)
-    maxPriority = 1; // divide by 0 is bad
+  maxPriority |= !maxPriority; // divide by 0 is bad(increase to 1 if 0)
   Initialize_();
   rgn_->InitForSchdulng();
 
   SchedInstruction *waitFor = NULL;
   InstCount waitUntil = 0;
+  double maxPriorityInv = 1 / maxPriority;
```

Comment:
Do we know that micro-optimizations like this are helping (i.e. did benchmarks show it)?

Also, I think this can be marked `const`.

Reply:
This one in particular will.  I profiled ACO a while ago, and the fp division instructions were responsible for a significant portion of the execution time. There are 10 ants in an iteration.  previously there were 2 fp divisions/ant for a total of 20 divisions/iteration.  This change replaces 10 fp divisions with 1 fp division and 10 fp multiplications.